saleor/graphql/order/filters.py

```diff
@@ -79,12 +80,23 @@ def filter_status(qs, _, value):
 
 
 def filter_customer(qs, _, value):
-    qs = qs.filter(
-        Q(user_email__ilike=value)
-        | Q(user__email__trigram_similar=value)
-        | Q(user__first_name__trigram_similar=value)
-        | Q(user__last_name__trigram_similar=value)
-    )
+    try:
+        validate_email(value)
+        qs = qs.filter(Q(user_email__iexact=value) | Q(user__email__iexact=value))
+    except ValidationError:
+        try:
+            first_name, last_name = value.split()
```

Comment:
This means the assumption that you will always provide value in format `{first_name} {last_name}`. What if I provide it in reverse order? Or only last_name? Is it expected that we will call the previous filter in that case?

Reply:
> This means the assumption that you will always provide value in format {first_name} {last_name}. What if I provide it in reverse order?

Good catch, i will adjust it to also handle `{last_name} {first_name}`.

> Or only last_name? Is it expected that we will call the previous filter in that case?

It is not exactly previous filter, previous filter were doing trigram similar search, while new one is doing full text search using `ilike`. 
Handling it this way it is expected.
1. if email is valid then search by email,
2. if not, try to split `value` into first and last name and try match using those values aka full name match,
3. if there was problem splitting value (aka not a full name), search using whole `value`.

If value is `last_name` it will be handled by point 3.

This is different from previous filter which was doing just search using stemming.